src/main/java/Duke.java

```diff
@@ -1,10 +1,49 @@
+import java.io.IOException;
+
+import duke.Parser;
+import duke.Storage;
+import duke.TaskList;
+import duke.Ui;
+import duke.command.Command;
+import duke.exception.DukeCommandException;
+import duke.exception.DukeException;
+import duke.exception.DukeToDoException;
+
 public class Duke {
```

Comment:
Overall, the code quality was great besides a couple of minute details, and I discovered different encapsulations and OOP use cases. Keep up the good work!